tests/sentry/tasks/test_on_demand_metrics.py

```diff
@@ -362,14 +365,23 @@ def test_process_widget_specs(
             for mock_call in _set_cardinality_cache.mock_calls
         )
 
+    expected_state = ""
```

Comment:
I'm dropping the `expected_applicable` parameter from `assert_on_demand_model` and adding `expected_state`, thus, I move some internal logic to this place in this test.

The test I will be adding in my next PR uses `OnDemandExtractionState.DISABLED_NOT_APPLICABLE` which `assert_on_demand_model` does not support checking for before my changes in here.